src/Modules/Settings/Dnn.PersonaBar.Prompt/Prompt.Web/src/components/Input/index.jsx

```diff
@@ -16,154 +17,163 @@ class Input extends Component {
+                        props.busy(false);
+                    }, errorCallback.bind(this));
+            }
         }
-        this.setFocus(false);
     }
 
-    changeUserMode(tokens) {
-        let { props } = this;
+    displayCmdInfo(txt) {
+        const { props } = this;
         const { actions } = props;
-        if (!tokens && tokens.length >= 2) {
-            return;
-        }
-        let mode = null;
-        if (this.hasFlag("--mode")) {
-            mode = this.getFlag("--mode", tokens);
-        } else if (!this.isFlag(tokens[1])) {
-            mode = tokens[1];
+        actions.runLocalCommand("INFO", "\n" + txt + "\n", "cmd");
+    }
+
+    runCmd() {
+        const txt = this.readInput();
+        if(txt) {
+            this.displayCmdInfo(txt);
+            if(!this.runLocalCmd(txt))
+                this.runServerCmd(txt);
+            this.setFocus(false);
         }
-        if (mode) {
-            actions.changeUserMode({ UserMode: mode.toUpperCase() }, () => {
-                if (mode.toUpperCase() === "EDIT")
+    }
+
+    changeUserMode(tokens) {
+        const { props } = this;
+        const { actions } = props;
+        if (!tokens && tokens.length >= 2) return;
+
+        tokens = tokens.map(token => token.toUpperCase());
+
+        const [mode] = tokens.slice(-1);
+
+        if (mode && !this.isFlag(mode) && Mode[mode]) {
+            actions.changeUserMode({ UserMode: mode }, () => {
+                if (mode === "EDIT")
                     util.utilities.closePersonaBar();
                 window.parent.document.location.reload(true);
-            }, (error) => {
-                actions.runLocalCommand("ERROR", error);
-            });
+            }, (error) => actions.runLocalCommand("ERROR", error));
         } else {
             actions.runLocalCommand("ERROR", formatString(Localization.get("Prompt_FlagIsRequired"), "Mode"));
         }
     }
+
     isFlag(token) {
         return (token && token.startsWith('--'));
     }
+
     getFlag(flag, tokens) {
-        let token = null;
-        if (!tokens || tokens.length) {
-            return null;
-        }
-        for (let i = 1; i < tokens.length; i++) {
-            token = tokens[i];
-            // did we find the flag name?
-            if (this.isFlag(token) && (token.toUpperCase() === flag.toUpperCase())) {
-                // is there a value to be had?
-                if ((i + 1) < tokens.length) {
-                    if (!this.isFlag(tokens[i + 1])) {
-                        return tokens[i + 1];
-                    } else {
-                        // next token is a flag and not a value. return nothing.
-                        return null;
-                    }
-                } else {
-                    // found but no value
-                    return null;
-                }
-            }
-        }
-        // not found
-        return null;
+        if (!tokens || tokens.length === 0) return null;
```

Comment:
I'd rather we keep mandatory braces. Isn't it part of our eslint setup?

Actually, is prompt set up with eslint?

Reply:
Yes, we have eslint.